exercises/luhn/example.go

```diff
@@ -1,27 +1,28 @@
 package luhn
 
-const testVersion = 1
+import "strings"
 
-func Valid(n string) bool {
-	if len(n) == 1 {
-		return false
-	}
-	d := extract(n)
-	if len(d) == 0 {
+const testVersion = 2
+
+func Valid(id string) bool {
+
+	if len(strings.TrimSpace(id)) == 1 {
 		return false
 	}
-	last := len(d) - 1
-	return (check(d[:last])+d[last])%10 == 0
-}
 
-func extract(n string) []int {
-	d := make([]int, 0, len(n))
-	for _, r := range n {
+	d := make([]int, 0, len(id))
+
+	for _, r := range id {
+		if r != ' ' && (r < '0' || r > '9') {
```

Comment:
*personally*, I would strive to avoid the repetition of the '0' and '9', by saying `continue` if `r == ' '`, then the remaining cases are `if`/`else` of each other. Assuming I didn't get my logic wrong. Does it make sense?